File: 01gpio/05rc_analog/rc_analog.js
var onoff = require('onoff');

(function measure(){
	var count = 0;
	var pin_charge = new onoff.Gpio(18, 'in');
	var pin_measure = new onoff.Gpio(23, 'out');
	pin_measure.writeSync(0);
	setTimeout(function(){
		pin_charge.unexport();
		pin_measure.unexport();
		
		pin_charge = new onoff.Gpio(18, 'out');
		pin_measure = new onoff.Gpio(23, 'in');
		pin_charge.write(1, function(err){
			if(!err){
				while(pin_measure.readSync() === 0 ){
					count++;
				}
				console.log(count);
				pin_charge.unexport();
				pin_measure.unexport();
				measure();
			}
		});
	}, 10);
})();

File: 01gpio/05rc_analog/rc_analog.py
import RPi.GPIO as GPIO
import time

GPIO.setmode(GPIO.BCM)
PIN_CHARGE = 18
PIN_MEASURE = 23

GPIO.setmode(GPIO.BCM)

while True:
    # for discharge
    GPIO.setup(PIN_CHARGE, GPIO.IN) 
    GPIO.setup(PIN_MEASURE, GPIO.OUT)
    GPIO.output(PIN_MEASURE, False) 
    time.sleep(0.1) 
    
    # for charging and measuring
    GPIO.setup(PIN_MEASURE, GPIO.IN) 
    GPIO.setup(PIN_CHARGE, GPIO.OUT) 
    count = 0                       
    GPIO.output(PIN_CHARGE, True)  
    while not GPIO.input(PIN_MEASURE):  
        count = count + 1           
    print(count)
